src/realm/alloc_slab.cpp

```diff
@@ -827,6 +827,11 @@ ref_type SlabAlloc::attach_file(const std::string& path, Config& cfg, util::Writ
         if (REALM_UNLIKELY(cfg.read_only))
             throw InvalidDatabase("Read-only access to empty Realm file", path);
 
+        // exFAT does not allocate a unique id for the file until it's non-empty
+        // It must be valid at this point because File::get_unique_id() is used to distinguish mappings_for_file in
+        // the encryption layer
+        m_file.resize(1);
```

Comment:
Since we're about to call prealloc() anyway, couldn't that just be moved up above writing the header?

Reply:
That does indeed seem more efficient, updated.